app/models/user.rb

```diff
@@ -13,13 +13,21 @@ class User < ApplicationRecord
   has_many :collection_options, dependent: :destroy
   has_many :collection_messaging_options, -> { where(option_type: CollectionOption::EMAIL_MESSAGES) }, class_name: "CollectionOption", dependent: :destroy
 
-  has_many :collections_with_options, through: :collection_options, source: :collection
-  has_many :collections_with_messaging, through: :collection_messaging_options, source: :collection
+  has_many :collections_with_options, through: :collection_options, source: :group
+  has_many :collections_with_messaging, through: :collection_messaging_options, source: :group
```

Comment:
This may have cascading impacts, but the user should have groups not collections
```suggestion
  has_many :groups_with_options, through: :collection_options, source: :group
  has_many :groups_with_messaging, through: :collection_messaging_options, source: :group
```